### PR TITLE
Demos: add Inferrers

### PR DESCRIPTION
**Pull request by Wovchena**

### CODE REVIEW

File: demos/object_detection_demo/python/proposal.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+        log.warning('The "--masks" options works only for "-at==yolov4". Option will be omitted')
+
+    cap = open_images_capture(args.input, args.loop)
+
+    if args.adapter == 'openvino':
+        plugin_config = get_user_config(args.device, args.nstreams, args.num_threads)
+        inferer = OvInferrer(create_core(), args.model, device=args.device, plugin_config=plugin_config,
+            nireq=args.num_infer_requests)
+    elif args.adapter == 'ovms':
+        raise 'kek'
```
Comment (anzhella-pankratova) [inline demos/object_detection_demo/python/proposal.py:210]: What is it?

Comment (Wovchena) [reply demos/object_detection_demo/python/proposal.py:210]: I haven't implemented it

File: demos/classification_benchmark_demo/cpp/main.cpp [hunk 2]
```diff
@@ -205,11 +197,11 @@ int main(int argc, char *argv[]) {
                 }
         }
 
-        slog::info << *InferenceEngine::GetInferenceEngineVersion() << slog::endl;
-        InferenceEngine::Core core;
-        AsyncPipeline pipeline(std::unique_ptr<ModelBase>(new ClassificationModel(FLAGS_m, FLAGS_nt, FLAGS_auto_resize, labels)),
-            ConfigFactory::getUserConfig(FLAGS_d, FLAGS_l, FLAGS_c, FLAGS_nireq, FLAGS_nstreams, FLAGS_nthreads),
-            core);
+        slog::info << ov::get_openvino_version() << slog::endl;
+        ov::runtime::Core core;
```
Comment (vladimir-dudnik) [inline demos/classification_benchmark_demo/cpp/main.cpp:201]: do not use runtime namespace, it was already removed from OV

File: demos/common/cpp/models/CMakeLists.txt [hunk 3]
```diff
@@ -12,4 +12,4 @@ source_group("include" FILES ${HEADERS})
 
 add_library(models STATIC ${SOURCES} ${HEADERS})
 target_include_directories(models PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
-target_link_libraries(models PRIVATE ngraph::ngraph gflags ${InferenceEngine_LIBRARIES} utils opencv_core opencv_imgproc)
+target_link_libraries(models PRIVATE gflags ${InferenceEngine_LIBRARIES} utils opencv_core opencv_imgproc)
```
Comment (vladimir-dudnik) [inline demos/common/cpp/models/CMakeLists.txt:15]: is that good time to change to OpenVINO_Libraries in cmake?

File: models/public/efficientdet-d1-tf/model.yml [hunk 4]
```diff
@@ -117,7 +117,7 @@ postprocessing:
 model_optimizer_args:
   - --input_shape=[1,640,640,3]
   - --input=image_arrays
-  - --layout=image_arrays(NHWC)
+  # - --layout=image_arrays(NHWC)
```
Comment (vladimir-dudnik) [inline models/public/efficientdet-d1-tf/model.yml:120]: why layout option either removed or commented out?

File: demos/smart_classroom_demo/cpp/main.cpp [hunk 5]
```diff
@@ -1,30 +1,28 @@
-// Copyright (C) 2018-2022 Intel Corporation
+// Copyright (C) 2018-2019 Intel Corporation
```
Comment (vladimir-dudnik) [inline demos/smart_classroom_demo/cpp/main.cpp:1]: 2022?

File: demos/smart_classroom_demo/cpp/main.cpp [hunk 6]
```diff
@@ -390,11 +388,22 @@ std::map<int, int> GetMapFaceTrackIdToLabel(const std::vector<Track>& face_track
     return face_track_id_to_label;
 }
 
+bool checkDynamicBatchSupport(const InferenceEngine::Core& ie, const std::string& device)  {
+    try  {
+        if (ie.GetConfig(device, CONFIG_KEY(DYN_BATCH_ENABLED)).as<std::string>() != InferenceEngine::PluginConfigParams::YES)
```
Comment (vladimir-dudnik) [inline demos/smart_classroom_demo/cpp/main.cpp:393]: old config API?

**Review by vladimir-dudnik — CHANGES_REQUESTED** [review]

**Pull request closed without merging**